### ISSUE TITLE
tags not updating in sidebar

### ISSUE THREAD

**Issue by droidgoo** [ufb/Textmarker#116]
when opening a series of  previously marked websites, the first one will load the sidebar up with the correct history details, but upon loading a 2nd website, only the marks will update in the sidebar to the the correct history details for that site... the tag remains the same as the first site.

the work around is to switch between open tabs from the article to the history tab, and then back. upon returning the tag history for that article will be correct.

it would also be nice to have the tags save to the text file along with the other infos and marks... suggestion.

**Comment by ufb** [comment]
> the tag remains the same as the first site

So, you have multiple open tabs with articles and when you switch between tabs the tags section of the sidebar won't update? I tried different szearios just now, both adding new tags directly inside the sidebar and also on the history page - when i switch between tabs the tags are being updated accordingly.
But what I just saw is that sometimes the tags don't get updated correctly if i load another page into the same tab.
Could you provide the exact procedure that leads to described bahavior?

> have the tags save to the text file along with the other infos and marks

Seems like a good idea, added to the todo list. Up until a few weeks ago I didn't know if anyone is even using the download-to-text-file feature :) It's an old relic from the beginning of this extension that I haven't worked on for quite a while. So any user feedback on what "shape" of those text files would be most helpful is welcome.

**Comment by droidgoo** [comment]
my work flow is probably more chaotic than even i intend it to be, so my
USE CASE might be unanticipated... more on that below, but i'm curious
about this statement.

>>  adding new tags ....  and also on the history page

i will frequently hit ENTER to save a tag rather than clicking the arrow,
only later to find myself on the history page without a tag... if there is
a way to add the tag from the history page, i would like to know.  Can you
describe how you did  this?

Yes, that is exactly what i'm doing. I'll have the bookmarks library page
(firefox) open with saved bookmarks to review, and when i double click on
them in that window they open in the same tab on my browser (default)
unless i right click and ask for a separate tab.... so what happens in the
sidebar is the Marks and Notes will update to match the new link, but the
tag will not refresh -- remaining the same as the last link i had open
there. The work around then is to a) either change the default behavior of
the browser, b) remember to always open each bookmark in a new tab, and
close the other tab, or c) click over to the history tab (which i usually
have open anyway), then back to the tab with the new bookmark which seems
to trigger the ENTIRE sidebar to refresh.

Neither a) or b) are desirable because it clutters up the browser and makes
keeping my open tabs organized even more difficult. So, c) is where i'm at.

>> Seems like a good idea, added to the todo list.

Thanks, the ability to save these marks between page loads is a worthwhile
endeavor and a boon to anyone trying to pull bits of info from many sources
in order to research a subject... so this naturally leads to the need to
collect these marks into a single convenient file. However, because the
file doesn't include the tags info, i'm forced to make separate files for
each tag... it's a bit cumbersome.

I also like the ability to archive a set of histories so i can document my
work as i go without having to wade thru older histories that i've already
dealt with. I haven't really used this feature yet, but i'm assuming it
just creates the .json file which i can rename to reflect that point in my
research for later recall if need be, and then i'm free to delete the
histories from the tab.





My thots on "shape":

Include tags, obviously, but there is also 2 flavors of Notes as well...
one flavor you have covered (the notes that go with the marks) but the
General Page notes don't get reported anywhere. Perhaps an option for that
as well in the drop down UX for the file options. Perhaps even a different
UX for that step would be more intuitive (like a check all to include or
somesuch).

As to format of the text file, what you have is already pretty good, but
i've made some observations:

There exists an line of dashes "----" that only gets included when you
choose the "w notes" option but does NOT get included when the just marks
option is selected. I could go either way but i'm not sure why they are
different.

As i said before the order of both the history and the marks within each
history are reversed in the file from what shows up on the history tab
sort, or in the detail windows and should probably match up to avoid
confusion... i would like to think of the history sort order and the detail
view windows to be more like a styled up preview of the text file.

the "URL: " bit of string on the front end of the link doesn't play well
with double clicking the line and pasting into a browser or getting read by
machine since it wrecks the link for for that use.. i have to strip off the
first 5 characters to use it.

I like that the title line is a clean header on its own line, but i wish it
was a little longer 100 - 128 characters ... it's often truncated too soon
for the long titles of things i'm looking at.

**Comment by ufb** [comment]
> i will frequently hit ENTER to save a tag rather than clicking the arrow,
> only later to find myself on the history page without a tag..

Currently only saves on button click, not on enter key

if there is a way to add the tag from the history page, i would like to know.  Can you
describe how you did  this?

![tagging](https://user-images.githubusercontent.com/17272252/81072876-1a6a8700-8ee7-11ea-90ea-4641bae1fe05.jpg)

> Yes, that is exactly what i'm doing

A bug that slid thru. Textmarker is listening to some native change events of the tabs.  On URL change the sidebar is being updated - but apparently not the tag section..

> I haven't really used this feature yet, but i'm assuming it
just creates the .json file which i can rename to reflect that point in my
research for later recall if need be, and then i'm free to delete the
histories from the tab.

Yep. You can name the .json file however you like. On import Textmarker is checking its size, format and if its JSON-parsable. The name is irrelevant;)
On import please make sure that you don't have history entries with the same name as the imported entries -> the imported ones will be ignored then.

> Include tags, obviously, but there is also 2 flavors of Notes as well...
one flavor you have covered (the notes that go with the marks) but the
General Page notes don't get reported anywhere. Perhaps an option for that
as well in the drop down UX for the file options. Perhaps even a different
UX for that step would be more intuitive (like a check all to include or
somesuch).

What do you mean by "drop down UX for the file options"?
I implemented the general page notes upon user request via e-mail. They are completely independent of your history entries. It's been a while, but if I remember correctly your general page notes should persist even if you delete the history entry for that URL. So you could have either general page notes OR highlights+notes OR both. But since both general page notes and history entries have the corresponding URL as identifier those 2 could be matched up for adding the general page notes to a text file download of your highlights. I'll keep it in the back of my head..

> There exists an line of dashes "----" that only gets included when you
choose the "w notes" option but does NOT get included when the just marks
option is selected.

Hehe, OK - so I give you the power to decide for either one of them options, I'm agnostic;)

> As i said before the order of both the history and the marks within each
history are reversed in the file from what shows up on the history tab
sort, or in the detail windows and should probably match up to avoid
confusion

Yep, I got that one. See [#114](https://github.com/ufb/Textmarker/issues/114), [#115](https://github.com/ufb/Textmarker/issues/114)

>  i would like to think of the history sort order and the detail
view windows to be more like a styled up preview of the text file.

Definately not. The text file download is just a side feature. The history page is the main source of truth for reflecting your highlights. Different use cases should be derived from this history (including the natural chronological order of highlights). Styling and different orders should be seen as variations of representation of this source.
In future releases I'm going to add more of those representational options - i.e. copying the highlights directly from the sidebar (I think that's another user request). So, copying to the clipboard, downloading as text file, downloading as .json - those will be extended and improved (for shaping the text file content I'll implement some of your suggestions (adding tags, fixing the order...))

> the "URL: " bit of string on the front end of the link doesn't play well
with double clicking the line and pasting into a browser or getting read by
machine

Interesting point. Noted;)
When re-shaping the text file content I'll have to make sure that it won't conflict with use cases of other users that might rely on the shape being as it is. Will have to think about it..

> I like that the title line is a clean header on its own line, but i wish it
was a little longer 100 - 128 characters ... it's often truncated too soon
for the long titles of things i'm looking at.

The title is not being truncated. Do you mean the history entry name? That would be the first line in the text file when including meta data. The name is being truncated, yes.

Sidenote: I'm not sure when I will be able to continue with development for Textmarker, it depends on the job situation in the next months. So, it's well possible that I have to postpone it for several months.

**Comment by droidgoo** [comment]
that's for the visual, pretty hard to miss now that i see it ... being at
the top of the list must have threw me ;)

much appreciate all the consideration and guidance you have provided and
hope your job situ sorts more in your favor than mine.

i'll just answer your questions below and wish you well.

>> What do you mean by "drop down UX for the file options"?

i'm referring to that drop down list that appears to the left when you
select "Save to text file".. there are a number of pre-defined combinations
that might be better represented by a "check all to include" sort of UX...
that way the user can configure what to leave in or drop out of the text
file.

>>  Do you mean the history entry name

Yes, that header bit that is editable from the history tab, that's what i'm
referring to.

**Comment by ufb** [comment]
Thanks for clarifications

> might be better represented by a "check all to include" sort of UX...
that way the user can configure what to leave in or drop out of the text
file.

Ah, got it - that appears to be a smart way to handle it!

> Yes, that header bit that is editable from the history tab, that's what i'm
> referring to.

Ah right, that's the name of the history entry. "title" is referring to the page's title. The former can only have 70 characters, the latter is unlimited I believe. There were several reasons for a restriction to 70 characters. Do you give a lot of names with more than 70 chars? Or you just use the page's title as the entry's name (which would be the default option)? Anyway, I might consider increasing the max chars - didn't think 70 was a too strict limitation ;)

> much appreciate all the consideration and guidance you have provided and
hope your job situ sorts more in your favor than mine.

My pleasure, and I wish you good luck as well!

**Comment by droidgoo** [comment]
yes, i just use the default to name the histories because it nearly always
reflects the headline of the article i'm reviewing.
